zellij-tile/src/lib.rs

```diff
@@ -3,10 +3,13 @@ pub mod prelude;
 pub mod shim;
 
 use data::*;
+use serde::de::DeserializeOwned;
 
 #[allow(unused_variables)]
 pub trait ZellijPlugin {
-    fn load(&mut self) {}
+    type Options: Default + DeserializeOwned;
+
+    fn load(&mut self, _: Self::Options) {}
```

Comment:
We don't need to use `_` here, we should probably use a more descriptive name for documentation purposes (something like `config` or `options`?) Even if we aren't using it, I've already added `#[allow(unused_variables)]` to suppress that particular warning.